pkg/operation/botanist/controlplane.go

```diff
@@ -995,15 +995,19 @@ func (b *Botanist) DeployKubeAPIServer(ctx context.Context) error {
 		foundDeployment = false
 	}
 
-	if b.ShootedSeed != nil && !hvpaEnabled {
+	if b.ShootedSeed != nil {
 		var (
 			apiServer  = b.ShootedSeed.APIServer
 			autoscaler = apiServer.Autoscaler
 		)
-		defaultValues["replicas"] = *apiServer.Replicas
 		minReplicas = *autoscaler.MinReplicas
 		maxReplicas = autoscaler.MaxReplicas
+	}
 
+	if b.ShootedSeed != nil && !hvpaEnabled {
+		apiServer := b.ShootedSeed.APIServer
+
+		defaultValues["replicas"] = *apiServer.Replicas
```

Comment:
```suggestion

		if !hvpaEnabled {
			defaultValues["replicas"] = *apiServer.Replicas
```

You can have the `if b.ShootedSeed != nil` only once

Reply:
No, then the else below changes its meaning. The else also needs to respect the case when ShootedSeed == nil (and hvpa whatever).

Reply:
The else on && connected ifs are super confusing I think btw.